src/main/java/seedu/address/nusmods/DataFetcherManager.java

```diff
@@ -0,0 +1,141 @@
+package seedu.address.nusmods;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.HttpUtil;
+import seedu.address.commons.util.JsonUtil;
+import seedu.address.nusmods.exceptions.NusmodsException;
+
+/**
+ * Handles API requests made by GradPad to the NUSMods public API to retrieve module data.
+ */
+public class DataFetcherManager extends DataFetcher {
+    private static final Logger logger = LogsCenter.getLogger(DataFetcher.class);
+    private final HttpUtil httpUtil;
+    private final String dataFilePath;
+
+    DataFetcherManager() {
+        httpUtil = HttpUtil.SINGLETON;
+        dataFilePath = DataFetcher.DATA_FILE_PATH;
+    }
+
+    DataFetcherManager(HttpUtil httpUtil) {
+        this.httpUtil = httpUtil;
+        dataFilePath = DataFetcher.DATA_FILE_PATH;
+    }
+
+    DataFetcherManager(HttpUtil httpUtil, String filePath) {
+        this.httpUtil = httpUtil;
+        this.dataFilePath = filePath;
+    }
+
+    /**
+     * Fetches a list of all NUS module summaries, then fetches module info for all CS modules (only) and
+     * saves them to a JSON file locally.
+     *
+     * @throws NusmodsException if an error occurs while fetching data from NUSMods API or while saving data.
+     */
+    public void fetchAndSaveModules() throws NusmodsException {
+        List<ModuleSummary> moduleSummaries = fetchModuleSummaryList();
+        List<ModuleSummary> csModules = filterModuleSummaries(moduleSummaries, "CS");
+        Map<String, ModuleInfo> moduleInfoMap = generateModuleInfoMap(csModules);
```

Comment:
I'm guessing we have to add the other non-CS modules that are in the syllabus as well to the moduleInfoMap

Reply:
Oh yeah you're right. In that case, I'll let `filterModuleSummaries` take in a bunch of keywords to filter against. We can then define the keywords in a hardcoded constant depending on the list of required modules.